src/item.cpp

```diff
@@ -3252,9 +3252,17 @@ std::string item::tname( unsigned int quantity, bool with_prefix, unsigned int t
     // for portions of string that have <color_ etc in them, this aims to truncate the whole string correctly
     unsigned int truncate_override = 0;
 
-    if( ( damage() != 0 || ( get_option<bool>( "ITEM_HEALTH_BAR" ) && is_armor() ) ) && !is_null() &&
+    if( ( damage() != 0 || is_gun() || ( get_option<bool>( "ITEM_HEALTH_BAR" ) && is_armor() ) ) &&
```

Comment:
FYI, if you go through with these changes, you will need to adjust the tests.

Reply:
What does this change do?